From cb06bba458a279ca99a9fc7524091be0b2596658 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sat, 19 Jul 2014 04:25:40 -0400
Subject: [pci] Convert ioapic to be builtin only, not modular

Message-id: <20140719042539.19438.13648.stgit@gir.stowe>
Patchwork-id: 85990
O-Subject: [RHEL7.1 PATCH 28/81] PCI: Convert ioapic to be builtin only, not modular
Bugzilla: 1110895
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110895
Upstream Status: 774104399459fe4c4195bdc657a898e292334137

commit 774104399459fe4c4195bdc657a898e292334137
Author: Yinghai Lu <yinghai@kernel.org>
Date:   Thu Jan 2 16:05:57 2014 -0800

    PCI: Convert ioapic to be builtin only, not modular

    Convert pci/ioapic.c to be builtin only, with no module option, so we can
    support IO-APIC hotplug.  Also make it depend on X86_IO_APIC.

    [bhelgaas: changelog]
    Signed-off-by: Yinghai Lu <yinghai@kernel.org>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/Kconfig b/drivers/pci/Kconfig
index 092408b..13fbdff 100644
--- a/drivers/pci/Kconfig
+++ b/drivers/pci/Kconfig
@@ -105,9 +105,10 @@ config PCI_PASID
    If unsure, say N.
 
 config PCI_IOAPIC
- tristate "PCI IO-APIC hotplug support" if X86
+ bool "PCI IO-APIC hotplug support" if X86
  depends on PCI
  depends on ACPI
+ depends on X86_IO_APIC
  default !X86
 
 config PCI_LABEL
diff --git a/drivers/pci/ioapic.c b/drivers/pci/ioapic.c
index 1b90579..7d6b157 100644
--- a/drivers/pci/ioapic.c
+++ b/drivers/pci/ioapic.c
@@ -113,6 +113,10 @@ static struct pci_driver ioapic_driver = {
  .remove  = ioapic_remove,
 };
 
-module_pci_driver(ioapic_driver);
+static int __init ioapic_init(void)
+{
+ return pci_register_driver(&ioapic_driver);
+}
+module_init(ioapic_init);
 
 MODULE_LICENSE("GPL");
-- 
1.7.1